core/src/test/java/feign/RequestTemplateTest.java

```diff
@@ -243,6 +242,18 @@ public void resolveTemplateWithHeaderWithNestedJson() {
         .hasHeaders(entry("A-Header", Collections.singletonList(json)));
   }
 
+  @Test
+  public void shouldNotResolveTemplateWithHeaderWithJsonLiteral() {
+    String json = "{ \"foo\": \"bar\", \"hello\": \"world\"}";
+    RequestTemplate template = new RequestTemplate().method(HttpMethod.GET)
+        .headerLiteral("A-Header", "{aHeader}");
+
+    template = template.resolve(mapOf("aHeader", json));
+
+    assertThat(template)
+        .hasHeaders(entry("A-Header", Collections.singletonList("{aHeader}")));
+  }
+
```

Comment:
Makes sure that if a expression that is added as a literal will not be evaluated as an expression when trying to resolve.